Source/Core/Core/PowerPC/JitLLVM/Jit.h

```diff
@@ -0,0 +1,303 @@
+                                        SConfig::GetInstance().setting)
+
+class JitLLVM : public JitBase
+{
+private:
+	std::unordered_map<u32, LLVMModule*> m_mods;
+	llvm::Module* m_main_mod;
+	llvm::ExecutionEngine* m_engine;
+	LLVMMemoryManager* m_mem_manager;
+	LLVMNamedBinding* m_named_binder;
+
+	// Debug helper
+	bool m_debug_enabled;
+
+	// Currently emitting things
+	LLVMModule* m_cur_mod;
+	LLVMFunction* m_cur_func;
+	LLVMBinding* m_cur_binder;
+
+	LLVMFunction* GetCurrentFunc() { return m_cur_func; }
+
+	PPCAnalyst::CodeBuffer code_buffer;
+
+	llvm::Value* JumpIfCRFieldBit(LLVMFunction* func, int field, int bit, bool jump_if_set);
+	void reg_imm(LLVMFunction* func, u32 d, u32 a, bool binary, u32 value, llvm::Instruction::BinaryOps opc, bool Rc = false);
+
+	// Helpers
+	llvm::Value* ForceSingle(LLVMFunction* func, llvm::Value* val);
+	// Returns an ICmpGT value
+	llvm::Value* HasCarry(LLVMFunction* func, llvm::Value* a, llvm::Value* b);
+	void ComputeRC(LLVMFunction* func, llvm::Value* val, int crf, bool needs_sext = true);
+	llvm::Value* GetCRBit(LLVMFunction* func, int field, int bit, bool negate);
+	void SetCRBit(LLVMFunction* func, int field, int bit);
+	void SetCRBit(LLVMFunction* func, int field, int bit, llvm::Value* val);
+	void ClearCRBit(LLVMFunction* func, int field, int bit);
+
+	// LoadStore Helpers
+	llvm::Value* GetEA(LLVMFunction* func, UGeckoInstruction inst);
+	llvm::Value* GetEAU(LLVMFunction* func, UGeckoInstruction inst);
+	llvm::Value* GetEAX(LLVMFunction* func, UGeckoInstruction inst);
+	llvm::Value* GetEAUX(LLVMFunction* func, UGeckoInstruction inst);
+	llvm::Value* CreateReadU64(LLVMFunction* func, llvm::Value* address);
+	llvm::Value* CreateReadU32(LLVMFunction* func, llvm::Value* address);
+	llvm::Value* CreateReadU16(LLVMFunction* func, llvm::Value* address, bool needs_sext);
+	llvm::Value* CreateReadU8(LLVMFunction* func, llvm::Value* address, bool needs_sext);
+	llvm::Value* CreateWriteU64(LLVMFunction* func, llvm::Value* val, llvm::Value* address);
+	llvm::Value* CreateWriteU32(LLVMFunction* func, llvm::Value* val, llvm::Value* address);
+	llvm::Value* CreateWriteU16(LLVMFunction* func, llvm::Value* val, llvm::Value* address);
+	llvm::Value* CreateWriteU8(LLVMFunction* func, llvm::Value* val, llvm::Value* address);
+
+	// Paired Loadstore helpers
+	llvm::Value* Paired_GetEA(LLVMFunction* func, UGeckoInstruction inst);
+	llvm::Value* Paired_GetEAU(LLVMFunction* func, UGeckoInstruction inst);
+	llvm::Value* Paired_GetEAX(LLVMFunction* func, UGeckoInstruction inst);
+	llvm::Value* Paired_GetEAUX(LLVMFunction* func, UGeckoInstruction inst);
+	llvm::Value* CreatePairedLoad(LLVMFunction* func, llvm::Value* paired, llvm::Value* type, llvm::Value* scale, llvm::Value* address);
+	llvm::Value* Paired_GetLoadType(LLVMFunction* func, llvm::Value* gqr);
+	llvm::Value* Paired_GetLoadScale(LLVMFunction* func, llvm::Value* gqr);
+
+public:
+	JitLLVM() : code_buffer(32000) {}
```

Comment:
Outside of the scope of this PR. All other recompilers use the same magic number.